packages/@aws-cdk/aws-lambda/lib/dlq.ts

```diff
@@ -0,0 +1,21 @@
+import { IFunction } from './function-base';
+
+/**
+ * A destination configuration
+ */
+export interface DlqDestinationConfig {
+  /**
+   * The Amazon Resource Name (ARN) of the destination resource
+   */
+  readonly destination: string;
+}
+
+/**
+ * A DLQ for an event source
+ */
+export interface IEventSourceDlq {
+  /**
+   * Returns the DLQ destination config of the DLQ
+   */
+  bind(target: IFunction): DlqDestinationConfig;
```

Comment:
Let's pass in both the `EventSourceMapping` object and the `IFunction`. One will not be needed in your current implementation of SNS and SQS, but opens up the possibility to build more complex DLQs.

Reply:
had to go up to IResource to get to an interface.  But a quick look at what's provided along the hierarchy makes me feel alright about that.  Thoughts?